library/bdBank/Model/Bank.php

```diff
@@ -74,11 +76,25 @@ public function canRefund(array $transaction, array $viewingUser = null)
         return false;
     }
 
-    public function getActionBonus($action, $extraData = array())
+    public function getActionBonus($action, $extraData = array(), $actionDate = null)
```

Comment:
Let's make `$actionDate` the second parameter without default value -> force all bonus code to provide a valid date. Maybe throw exception if action date is empty / not a number too.

Reply:
ok anh

Reply:
ah but this signature changing can affect other addons anh, because current second parameter `$extraData` is optional

Reply:
Yeah, it's the point. This is a major / breaking change so all dependencies should be updated to work correctly or stop working.

Reply:
hmm looks like this won't work for attachments. Currently, we do not make transactions for individual attachments (`bdBank_Model_Bank#macro_bonusAttachment`).  
While it also does not make sense to use the creation date of the content (that the attachments belong to) as the actionDate, does it?

Reply:
Let's drop attachment bonuses completely then.

Reply:
done via https://github.com/xfrocks/bdBank/pull/4/commits/d8c34a8b4825220f0312b99e205f347515520321